dbms/src/Storages/MergeTree/MergeTreeDataMergerMutator.cpp

```diff
@@ -1185,6 +1200,10 @@ MergeTreeDataMergerMutator::MergeAlgorithm MergeTreeDataMergerMutator::chooseMer
     if (need_remove_expired_values)
         return MergeAlgorithm::Horizontal;
 
+    for (const auto & part : parts)
+        if (!part->supportsVerticalMerge())
```

Comment:
Maybe we can additionally make some changes in MergeSelector, so it will try to select to merge parts with the same type?